ui/appui-react/src/appui-react/layout/widget-panels/Panel.tsx

```diff
@@ -59,6 +60,19 @@ export function WidgetPanel() {
   const side = React.useContext(PanelSideContext);
   const draggedPanelSide = React.useContext(DraggedPanelSideContext);
   assert(!!side);
+  const { enabled: previewHorizontalPanelAlignmentEnabled, alignments } =
+    usePreviewHorizontalPanelAlign();
+
+  const previewHorizontalPanelAlignAttributes = React.useMemo(
+    () =>
+      previewHorizontalPanelAlignmentEnabled
+        ? {
+            "data-preview-horizontal-panel-align-top": alignments.top,
+            "data-preview-horizontal-panel-align-bottom": alignments.bottom,
+          }
+        : {},
+    [alignments.bottom, alignments.top, previewHorizontalPanelAlignmentEnabled]
+  );
```

Comment:
This feature is introduced as a parallel implementation to the existing "span" feature for the top and bottom panel because "span" is saved to the local storage state, and if the preview feature is disabled, it will cause the last state to stick and not be changeable. (And the current implementation only supports `true`/`false`, for `justify`/`center`, so I did not want to change this part, it was simpler/safer to build it in a parallel system)